src/bin/download_sysext.rs

```diff
@@ -113,20 +121,26 @@ impl<'a> Package<'a> {
         }
     }
 
-    fn verify_signature_on_disk(&mut self, from_path: &Path, pubkey_path: &str) -> Result<(), Box<dyn Error>> {
+    fn verify_signature_on_disk(&mut self, from_path: &Path, pubkey_path: &str) -> Result<Vec<u8>, Box<dyn Error>> {
         let upfile = File::open(from_path)?;
 
+        // create a BufReader to pass down to parsing functions.
+        let upfreader = &mut BufReader::new(upfile);
+
         // Read update payload from file, read delta update header from the payload.
-        let res_data = fs::read_to_string(from_path);
+        let header = delta_update::read_delta_update_header(upfreader)?;
 
-        let header = delta_update::read_delta_update_header(&upfile)?;
+        let mut delta_archive_manifest = delta_update::get_manifest_bytes(upfreader, &header)?;
 
         // Extract signature from header.
-        let sigbytes = delta_update::get_signatures_bytes(&upfile, &header)?;
+        let sigbytes = delta_update::get_signatures_bytes(upfreader, &header, &mut delta_archive_manifest)?;
+
+        // Extract data blobs.
+        let datablobs = delta_update::get_data_blobs(upfreader, &header, &delta_archive_manifest)?;
 
-        // Parse signature data from the signature containing data, version, special fields.
-        let _sigdata = match delta_update::parse_signature_data(res_data.unwrap().as_bytes(), &sigbytes, pubkey_path) {
-            Some(data) => data,
+        // Parse signature data from sig blobs, data blobs, public key.
+        let _ = match delta_update::parse_signature_data(&sigbytes, datablobs.as_slice(), pubkey_path) {
+            Some(sigdata) => sigdata,
```

Comment:
Instead of returning something here that's thrown away it rather makes sense to use a regular match statement without `let _ = ` in front of it.

Reply:
Simply removed both `let _ = ` and `sigdata`.